src/game_api/state.hpp

```diff
@@ -64,12 +64,27 @@ struct StateMemory
     uint32_t loading;
     /// The global level illumination, very big and bright.
     Illumination* illumination;
-    /// Current fade-to-black amount (0.0 = all visible; 1.0 = all black). Manipulated by the loading routine when loading > 0.
-    float fadevalue;
-    /// Amount of frames the fadeout should last when loading
-    uint32_t fadeout;
-    /// Amount of frames the fadein should last when loading
-    uint32_t fadein;
+    union
+    {
+        /// Current fade-to-black amount (0.0 = all visible; 1.0 = all black). Manipulated by the loading routine when loading > 0.
+        float fade_value;
+        /// NoDoc
+        float fadevalue;
+    };
+    union
+    {
+        /// Remaining frames for fade-in/fade-out when loading. Counts down to 0.
+        uint32_t fade_timer;
+        /// NoDoc
+        uint32_t fadeout;
+    };
+    union
+    {
+        /// Total frames for fade-in/fade-out when loading.
+        uint32_t fade_length;
+        /// NoDoc
+        uint32_t fadein;
+    };

```

Comment:
I would just rename them, then in `state_lua.cpp` you can do the NoDoc

Reply:
Yep you can remove the unions, change all the internal references to the new names and add the deprecated lua bindings with NoDoc. There are probably examples somewhere.